lib/auth/auth.go

```diff
@@ -3057,6 +3061,23 @@ func (a *Server) ListResources(ctx context.Context, req proto.ListResourcesReque
 	return a.Cache.ListResources(ctx, req)
 }
 
+// AppendTraceConnectionDiagnostics adds a Trace into the ConnectionDiagnostics.
+// This is a convinience method that calls GetConnectionDiagnostic, appends the trace and updates the ConnectionDiagnostic.
+func (a *Server) AppendTraceConnectionDiagnostic(ctx context.Context, name string, t *types.ConnectionDiagnosticTrace) (types.ConnectionDiagnostic, error) {
+	connectionDiagnostic, err := a.GetConnectionDiagnostic(ctx, name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	connectionDiagnostic.AppendTrace(t)
```

Comment:
This is not concurrency-safe as mentioned above unless Update uses compare-and-swap.

Reply:
I'll change the call to do a compare-and-swap
Thank you :pray: